src/PrestaShopBundle/Controller/Admin/Sell/Catalog/Product/CombinationController.php

```diff
@@ -348,23 +347,17 @@ public function getListAction(int $productId, ProductCombinationFilters $combina
     /**
      * @AdminSecurity("is_granted('update', request.get('_legacy_controller'))")
      *
-     * @param int $productId
      * @param ProductCombinationFilters $filters
      *
      * @return JsonResponse
      */
-    public function getCombinationIdsAction(int $productId, ProductCombinationFilters $filters): JsonResponse
+    public function getCombinationIdsAction(ProductCombinationFilters $filters): JsonResponse
     {
-        /** @var CombinationRepository $repository */
-        $repository = $this->get('prestashop.adapter.product.combination.repository.combination_repository');
-
-        $combinationIds = $repository->getCombinationIds(new ProductId($productId), $filters);
-        $data = [];
-        foreach ($combinationIds as $combinationId) {
-            $data[] = $combinationId->getValue();
-        }
+        $combinationIds = $this->getQueryBus()->handle(new GetCombinationIds($filters));
```

Comment:
You will need to transform the `$filters` a bit here since your query requires:
- **productId**
- **shopConstraint**
- the other options like filters are optional but should be extracted from the $filters object